backend/controllers/profile.controller.js

```diff
@@ -2,11 +2,11 @@ const User = require('../models/user.model.js');
 
 exports.getUserProfile = async (req, res) => {
   try {
-    const user = await User.findById(res.locals.userId).select('-password');
```

Comment:
can you please keep the "res.locals.userId", it's already implemented and it's working. The value comes from authentication middleware